lib/puppet/provider/package/pkgng.rb

```diff
@@ -0,0 +1,145 @@
+require 'puppet/provider/package'
+
+Puppet::Type.type(:package).provide :pkgng, :parent => Puppet::Provider::Package do
+  desc "A PkgNG provider for FreeBSD and DragonFly."
+
+  commands :pkg => "/usr/local/sbin/pkg"
+
+  confine :operatingsystem => [:freebsd, :dragonfly]
+  confine :pkgng_enabled => :true
+
+  defaultfor :operatingsystem => :freebsd
+  defaultfor :pkgng_enabled => :true
```

Comment:
That sounds reasonable.  I'll try to work out a test for this and include that work in the commit.